lib/lambda_ethereum_consensus/beacon/sync_blocks.ex

```diff
@@ -0,0 +1,98 @@
+defmodule LambdaEthereumConsensus.Beacon.SyncBlocks do
+  @moduledoc """
+    Performs an optimistic block sync from the finalized checkpoint to the current slot.
+  """
+
+  use Task
+
+  require Logger
+
+  alias LambdaEthereumConsensus.Beacon.PendingBlocks
+  alias LambdaEthereumConsensus.ForkChoice.Store
+  alias LambdaEthereumConsensus.P2P.BlockDownloader
+  alias LambdaEthereumConsensus.StateTransition.Misc
+
+  @blocks_per_chunk 20
+
+  @type chunk :: %{from: SszTypes.slot(), count: integer()}
+
+  def start_link(opts) do
+    Task.start_link(__MODULE__, :run, [opts])
+  end
+
+  def run(_opts) do
+    # TODO: this is a hack to make sure the on_tick has been called at least once
+    Process.sleep(2000)
```

Comment:
This shouldn't be needed since `on_tick` is called on `Store` initialization

Reply:
awesome!